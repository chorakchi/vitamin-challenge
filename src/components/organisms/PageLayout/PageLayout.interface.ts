import { HTMLAttributes } from "react";
export interface PageLayoutProps
  extends HTMLAttributes<HTMLElement> {
  title?: string;
  description?: any;
  onChangeRoute?: (path: string) => void;
  currentPath?: string;
}
