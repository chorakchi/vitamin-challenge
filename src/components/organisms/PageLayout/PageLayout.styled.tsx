import styled from "styled-components";

export const StyledWrapper = styled.div`
padding: 5%;
margin: 0;
height: 100%;
width: 100%;



@media screen and (max-width: 1280px) {
 
    gap: 18px;
    padding: 2%;
    padding-bottom: 120px;
}
}

.mainBlock{
    width: 100%;
    height: 100%;
    gap: 18px;
    padding-top: 32px;
    @media screen and (max-width: 1280px) {
        flex-direction: column;
gap:20px;

}
`;
