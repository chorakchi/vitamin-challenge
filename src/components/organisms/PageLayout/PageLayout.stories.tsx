import { ComponentStory, ComponentMeta } from "@storybook/react";

import { PageLayout } from "./PageLayout";
import {mockData } from './PageLayout.mock'

export default {
  title: "Organisms/PageLayout",
  component: PageLayout,
} as ComponentMeta<typeof PageLayout>;

const Template: ComponentStory<typeof PageLayout> = (args) => (
  <PageLayout {...args} />
);

export const PageLayout_ = Template.bind({});
PageLayout_.args = {
  onSelect:(id)=>{alert(id)},
  ...mockData
}