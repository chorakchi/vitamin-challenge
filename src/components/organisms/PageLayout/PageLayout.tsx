import { StyledWrapper } from "./PageLayout.styled";
import { PageLayoutProps } from "./PageLayout.interface";
import { Stack, Typography } from "components";
export const PageLayout: React.FC<PageLayoutProps> = ({
  children,
  title,
  description,
  onChangeRoute,
  currentPath,
  ...props
}: PageLayoutProps) => {
  return (
    <StyledWrapper {...props}>
      <Stack direction="column" className="">
        <Typography variant="headline_xl">{title}</Typography>
        <Typography>{description}</Typography>
        <Stack direction="row" className="mainBlock">
          {children}
        </Stack>
      </Stack>
    </StyledWrapper>
  );
};
