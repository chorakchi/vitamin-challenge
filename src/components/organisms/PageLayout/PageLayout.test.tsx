import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { PageLayout } from "./PageLayout";

describe("PageLayout", () => {
  it("should render the PageLayout", () => {
    const subject = shallow(<PageLayout />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer
      .create(<PageLayout onChangeRoute={() => {}} currentPath="" />)
      .toJSON();
    expect(subject).toMatchSnapshot();
  });
});
