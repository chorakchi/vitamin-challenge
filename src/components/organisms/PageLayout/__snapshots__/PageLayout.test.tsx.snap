// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout should render correctly and match to the Snapshot 1`] = `
<div
  className="PageLayoutstyled__StyledWrapper-sc-1kudg7l-0 bZBWEP"
>
  <div
    className="Stackstyled__StackStyled-sc-1adxq1u-0 bnrRyH"
    direction="column"
  >
    <div
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 jzzjid"
    />
    <div
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
    />
    <div
      className="Stackstyled__StackStyled-sc-1adxq1u-0 jWmaRX mainBlock"
      direction="row"
    />
  </div>
</div>
`;
